### PR TITLE
Add convenience function to save compressed zarr

### PR DESCRIPTION
**Pull request by observingClouds**
This PR adds the `to_compressed_zarr` accessor and applies a standard compression algorithm with bit shuffling by default. Other compression algorithms can be given as options.

- [x] create tests

### CODE REVIEW

**Comment by observingClouds** [comment]
Do we want to have a `save_compression.py` with both (NetCDF and Zarr) accessors or a `save_compression_nc.py` and `save_compression_zarr.py`? I probably prefer to have them both in one file. Any opinions @aaronspring ?

**Comment by aaronspring** [comment]
> I  probably prefer to have them both in one file.

me too

**Comment by aaronspring** [comment]
@observingClouds just out of curiosity: wasnt there an even easier way with zarr, where `xr_bitround` doesnt even need to be called? Or was this the case where some compression attrs was still saved in the file that made it a bit tricky for interoperability later?

**Comment by observingClouds** [comment]
Yes, one could also do something like

```python
import numcodecs
encoding = {
    "var1": {
        "filters": [numcodecs.BitRound(keepbits=14)],
        "compressor": numcodecs.Blosc("zstd",shuffle=numcodecs.Blosc.BITSHUFFLE)
    }
}

ds.to_zarr("out.zarr", encoding=encoding)
```
But as you mentioned, the issue with this solution is that "filters" are saved in the zarr files (`.zattrs` and `.zmetadata`) which makes it necessary for users to have this filter installed on their machine when opening the dataset. However, this particular filter can be seen as a one way filter and does not need a decoder for reading.

Applying bitrounding directly to the Dataset circumvents the writing of the `filters` information, but still keeps the information about the number of keepbits in the `_QuantizeBitRoundNumberOfSignificantDigits` attribute.

**Comment by aaronspring** [comment]
great. maybe you can include this snippet somewhere in the documentation, or just in a doctest for reference

File: bitinformation_pipeline/save_compressed.py [hunk 1]
```diff
@@ -115,5 +118,83 @@ def __call__(
...
+    ds_bitrounded,
+    compressor=numcodecs.Blosc("zstd", shuffle=numcodecs.Blosc.BITSHUFFLE),
+):
+    """Generate encoding for ds_bitrounded.to_zarr(encoding).
+
+    Example:
+        >>> ds = xr.tutorial.load_dataset("rasm")
+        >>> get_compress_encoding_zarr(ds)
+        {'Tair': {'zlib': True, 'shuffle': True, 'complevel': 9, 'chunksizes': (36, 205, 275)}}
+        >>> get_compress_encoding_zarr(ds, for_cdo=True)
```
Comment (aaronspring) [inline bitinformation_pipeline/save_compressed.py:135]:
```suggestion
```

Comment (aaronspring) [reply bitinformation_pipeline/save_compressed.py:135]: for_cdo not needed

File: bitinformation_pipeline/save_compressed.py [hunk 2]
```diff
@@ -115,5 +118,83 @@ def __call__(
...
+        {'Tair': {'zlib': True, 'shuffle': True, 'complevel': 9, 'chunksizes': (1, 205, 275)}}
```
Comment (aaronspring) [inline bitinformation_pipeline/save_compressed.py:136]:
```suggestion
```

**Pull request merged**